### PR TITLE
Feature Headless - Category Listing Page

### PR DESCRIPTION
**Pull request by Souvik-Ignitiv**

### CODE REVIEW

File: components/catrgory-list-page/Category/ChildCategory.styles.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  categoryText: {
+    fontSize: '20px',
+    fontWeight: 'bold',
+  },
+  gridContainer: {
+    display: { md: 'flex', xs: 'block' },
+    marginTop: { xs: '0px', md: '0px' },
+    marginBottom: { xs: '15px', md: '15px' },
+    padding: '0',
+    margin: '0 -16px',
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.styles.ts]: Can we remove negative margin ?

File: components/catrgory-list-page/Category/ChildCategory.styles.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  categoryItem: {
+    background: theme?.palette.primary.light,
+    height: '350px',
+    borderRadius: '5px',
+    marginBottom: '20px',
+    overflow: 'hidden',
+    margin: '0 1%',
+    textAlign: 'center',
+    '&:hover': {
+      backgroundColor: '#ffffff',
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.styles.ts]: 1. Use color hex codes from theme file. If it's not available then add it in theme.
2. Make above change in whole file

File: components/catrgory-list-page/Category/ChildCategory.styles.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    fontWeight: 'bold',
+    fontSize: '44px',
+    lineHeight: '1.2',
+  },
+  shopAllButton: {
+    padding: '10px 20px',
+    borderRadius: '50px',
+    marginRight: '15px',
+    backgroundColor: theme?.palette.primary.main,
+    border: '1px solid ' + theme?.palette.primary.main,
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.styles.ts]: Replace **'1px solid ' + theme?.palette.primary.main** with **`1px solid ${theme?.palette.primary.main}`**

File: components/catrgory-list-page/Category/ChildCategory.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  Typography,
+  Button,
+} from '@mui/material'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+
+import { ChildCategoryStyle } from './ChildCategory.styles'
+import { categorySearchGetters, productGetters } from '@/lib/getters'
+import { uiHelpers } from '@/lib/helpers'
+interface Item {
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.tsx:24]: Add 1 line space after import statements

File: components/catrgory-list-page/Category/ChildCategory.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+  const [items, setItems] = useState(categories)
+  const [selectedItems, setSelectedItems] = useState<Item[]>([])
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const isMobile = useMediaQuery(kiboTheme.breakpoints.down('sm'))
+  const isTablet = useMediaQuery(kiboTheme.breakpoints.down('md'))
+  const itemsPerPage = isMobile ? 24 : isTablet ? 24 : 24
+  const router = useRouter()
+  const { t } = useTranslation('common')
+  const handleItemClick = (item: Item) => {
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.tsx:71]: Add 1 line space after useTranslation hook

File: components/catrgory-list-page/Category/ChildCategory.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      nextIndex + itemsPerPage >= items.length ? 0 : nextIndex + itemsPerPage
+    )
+  }
+
+  useEffect(() => {
+    setItems(categories)
+  }, [categories])
+
+  const isPrevDisabled = currentIndex === 0
+  const isNextDisabled = currentIndex + itemsPerPage >= items.length
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.tsx]: Move constants to the top

File: components/catrgory-list-page/Category/ChildCategory.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+  const handleNextClick = () => {
+    setCurrentIndex((nextIndex) =>
+      nextIndex + itemsPerPage >= items.length ? 0 : nextIndex + itemsPerPage
+    )
+  }
+
+  useEffect(() => {
+    setItems(categories)
+  }, [categories])
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.tsx:93]: Move useEffect before functions

File: components/catrgory-list-page/Category/ChildCategory.tsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+  const isPrevDisabled = currentIndex === 0
+  const isNextDisabled = currentIndex + itemsPerPage >= items.length
+
+  return (
+    <Container maxWidth={'xl'} sx={ChildCategoryStyle.container}>
+      <Box>
+        <Box sx={ChildCategoryStyle.navigationContainer}>
+          <Typography variant="h1" sx={ChildCategoryStyle.categoryName}>
+            {childCategory?.content?.name}
```
Comment (bsatignitiv) [inline components/catrgory-list-page/Category/ChildCategory.tsx]: 1. Create a getter function to get any data from childCategory object 
2. Make this change everywhere

File: components/catrgory-list-page/HeroBanner/ClpHeroBanner.styles.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export const ClpHeroBannerStyle = {}
```
Comment (bsatignitiv) [inline components/catrgory-list-page/HeroBanner/ClpHeroBanner.styles.ts]: Remove this file if we have not added any styles

File: components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  clpHeroBannerProps: ClpHeroBannerProps
+}
+interface ClpHeroBannerProps {
+  imageUrl: string
+  mobileImageUrl: string
+  imageAlt: string
+  description: string
+}
+
+const MainStyle = styled('div')({
```
Comment (bsatignitiv) [inline components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx]: What do you mean by MainStyle ? Give it a meaningful name.

File: components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+const ClpHeroBanner = ({ clpHeroBannerProps }: ItemProps) => {
+  const kiboTheme = useTheme()
+  const mobileView = useMediaQuery(kiboTheme.breakpoints.down('sm'))
+
+  const { description, imageUrl, mobileImageUrl, imageAlt } = clpHeroBannerProps || {}
+
+  return (
+    <>
+      {clpHeroBannerProps && (
+        <>
```
Comment (bsatignitiv) [inline components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx]: This fragment (<>) is not required. We can remove it.

File: src/pages/clp/[categoryCode].tsx [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+const { publicRuntimeConfig } = getConfig()
+const apiKey = publicRuntimeConfig?.builderIO?.apiKey
+
+builder.init(apiKey)
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const { locale } = context
+
+  const pageUrl = context.resolvedUrl
+  console.log('context', pageUrl, context.query.categoryCode)
```
Comment (bsatignitiv) [inline src/pages/clp/[categoryCode].tsx]: Remove console

**Review by bsatignitiv — CHANGES_REQUESTED** [review]

File: components/catrgory-list-page/Category/ChildCategory.styles.ts (hunk 1 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/Category/ChildCategory.styles.ts]: removed negative margin

File: components/catrgory-list-page/Category/ChildCategory.styles.ts (hunk 2 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/Category/ChildCategory.styles.ts]: change all the color and added them from them.ts

Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/Category/ChildCategory.styles.ts]: theme.ts

File: components/catrgory-list-page/Category/ChildCategory.tsx (hunk 4 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/Category/ChildCategory.tsx:24]: space added

File: components/catrgory-list-page/Category/ChildCategory.tsx (hunk 5 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/Category/ChildCategory.tsx:71]: space added

File: components/catrgory-list-page/Category/ChildCategory.tsx (hunk 8 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/Category/ChildCategory.tsx]: created getter function to get data

File: components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx (hunk 10 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx]: change the name to HeroBanner Container

File: src/pages/clp/[categoryCode].tsx (hunk 12 above)
Comment (Souvik-Ignitiv) [reply src/pages/clp/[categoryCode].tsx]: removed console

File: components/catrgory-list-page/HeroBanner/ClpHeroBanner.styles.ts (hunk 9 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/HeroBanner/ClpHeroBanner.styles.ts]: removed the file

File: components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx (hunk 11 above)
Comment (Souvik-Ignitiv) [reply components/catrgory-list-page/HeroBanner/ClpHeroBanner.tsx]: remove line no 33 for 31 it was giving error

**Pull request merged**